compiler/rustc_mir_build/src/build/coverageinfo.rs

```diff
@@ -121,14 +116,9 @@ impl BranchInfoBuilder {
                 _ => {
                     // Do not generate mcdc mappings and statements for decisions with too many conditions.
                     let rebase_idx = self.mcdc_branch_spans.len() - decision.conditions_num + 1;
-                    let to_normal_branches = self.mcdc_branch_spans.split_off(rebase_idx);
-                    self.branch_spans.extend(to_normal_branches.into_iter().map(
-                        |MCDCBranchSpan { span, true_marker, false_marker, .. }| BranchSpan {
-                            span,
-                            true_marker,
-                            false_marker,
-                        },
-                    ));
+                    for branch in &mut self.mcdc_branch_spans[rebase_idx..] {
+                        branch.condition_info = None;
+                    }
```

Comment:
When the MC/DC code gives up on full MC/DC instrumentation, instead of creating a `BranchSpan`, it now creates an `MCDCBranchSpan` with `condition_info: None`.

That span will eventually get turned back into an ordinary branch region, in the coverage MIR pass. This change will let me improve how ordinary branch coverage is handled during MIR building, without having to also worry about how it interacts with MC/DC.